### PR TITLE
added changes for multilabel templates

### PR DESCRIPTION
**Pull request by kprokofi**
This PR connected with https://github.com/openvinotoolkit/model_preparation_algorithm/pull/66
* Updated submodule
* Added support for separate yaml config for multilabel classification
* Added attribute to runner for EMA Hook support

### CODE REVIEW

**Comment by kprokofi** [comment]
run ote_tests

**Comment by kprokofi** [comment]
run ote-test

File: external/model-preparation-algorithm/mpa_tasks/apis/classification/task.py [hunk 1]
```diff
@@ -280,11 +280,17 @@ def _add_predictions_to_dataset(self, prediction_results, dataset, update_progre
...
         warmup_iters = int(self._hyperparams.learning_parameters.learning_rate_warmup_iters)
-        lr_config = (
-            ConfigDict(warmup_iters=warmup_iters)
-            if warmup_iters > 0
-            else ConfigDict(warmup_iters=warmup_iters, warmup=None)
-        )
+        if self._multilabel:
+            # hack to use 1cycle policy
+            lr_config = (
+                ConfigDict(max_lr=self._hyperparams.learning_parameters.learning_rate, warmup=None)
```
Comment (supersoob) [inline external/model-preparation-algorithm/mpa_tasks/apis/classification/task.py:285]: I can't find update of learning rate for multi-label. Is it necessary now? Are you going to update soon?

Comment (kprokofi) [reply external/model-preparation-algorithm/mpa_tasks/apis/classification/task.py:285]: This is parameter `max_lr` needed for 1cycle policy. It set to current learning rate. Unfortunately, there is no other way to receive information about learning rate, but we need it to scheduler to work.
For now, there is no learning rate update. In the nearest future we will be dividing templates, but now we have unite learning rate parameter for both tasks.

**Comment by JihwanEom** [comment]
run ote-test

**Comment by sovrasov** [comment]
run ote-test

**Comment by kprokofi** [comment]
Tested manually on mclx80. Everything works as expected now. Checked all 3 models on Aerial Maritime with 2-3 training rounds. Also, tried class incremental setup.

**Review by sovrasov — DISMISSED** [review]

File: external/model-preparation-algorithm/mpa_tasks/apis/classification/task.py (hunk 1 above)
Comment (sovrasov) [reply external/model-preparation-algorithm/mpa_tasks/apis/classification/task.py:285]: It worth's noticing that now we can't change the scheduler for multilabel classification without changing this line because the other schedulers don't expect `max_lr` parameter. We can workaround this by letting each scheduler ignore irrelevant arguments, otherwise we have to change the overall design of the recipies initialization step to fix the mentioned problem.

**Comment by kprokofi** [comment]
I removed wth_empty_annoation() method. It is a problem with annotations changing ([CVS-88913](https://jira.devtools.intel.com/browse/CVS-88913), [CVS-93867](https://jira.devtools.intel.com/browse/CVS-93867), [CVS-93606](https://jira.devtools.intel.com/browse/CVS-93606))

Tested this on task chaining. Seems to fix the problem.

File: external/model-preparation-algorithm/configs/classification/efficientnet_b0_cls_incr/model_multilabel.py [hunk 2]
```diff
@@ -1,5 +1,5 @@
 _base_ = [
-    "../../../submodule/models/classification/ote_efficientnet_b0_multilabel.yaml",
+    "../../../submodule/models/classification/ote_efficientnet_b0_multilabel_al.yaml",
```
Comment (harimkang) [inline external/model-preparation-algorithm/configs/classification/efficientnet_b0_cls_incr/model_multilabel.py:2]: what is "al" means? (It is not a name that can be understood at a glance.)

File: external/model-preparation-algorithm/mpa_tasks/apis/classification/task.py [hunk 3]
```diff
@@ -133,7 +133,6 @@ def infer(
         logger.info("called infer()")
         stage_module = "ClsInferrer"
         self._data_cfg = self._init_test_data_cfg(dataset)
-        dataset = dataset.with_empty_annotations()
```
Comment (harimkang) [inline external/model-preparation-algorithm/mpa_tasks/apis/classification/task.py:136]: Does the 'ote eval' show the same score as before?

File: external/model-preparation-algorithm/mpa_tasks/apis/classification/task.py [hunk 4]
```diff
@@ -313,13 +318,13 @@ def _init_recipe(self):
         train_type = self._hyperparams.algo_backend.train_type
         logger.info(f"train type = {train_type}")
 
-        recipe = os.path.join(recipe_root, "class_incr.yaml")
```
Comment (harimkang) [inline external/model-preparation-algorithm/mpa_tasks/apis/classification/task.py:316]: I think it needs to be declared first, either None or as it is.

**Review by harimkang — COMMENTED** [review]
I left some comments.

If you request again after confirmation, I will approve.

File: external/model-preparation-algorithm/mpa_tasks/apis/classification/task.py (hunk 3 above)
Comment (kprokofi) [reply external/model-preparation-algorithm/mpa_tasks/apis/classification/task.py:136]: Yes, the result is the same. Checked it.

File: external/model-preparation-algorithm/configs/classification/efficientnet_b0_cls_incr/model_multilabel.py (hunk 2 above)
Comment (kprokofi) [reply external/model-preparation-algorithm/configs/classification/efficientnet_b0_cls_incr/model_multilabel.py:2]: it stands for "angular loss". I can change it to AAM. AAM - angular asymmetric margin loss, it is more consistent with our paper.

Comment (kprokofi) [reply external/model-preparation-algorithm/configs/classification/efficientnet_b0_cls_incr/model_multilabel.py:2]: But  we need again merge changes in MPA. I propose to leave it as is. al - angular loss

File: external/model-preparation-algorithm/mpa_tasks/apis/classification/task.py (hunk 4 above)
Comment (kprokofi) [reply external/model-preparation-algorithm/mpa_tasks/apis/classification/task.py:316]: please take a look

File: external/model-preparation-algorithm/mpa_tasks/apis/classification/task.py (hunk 1 above)
Comment (supersoob) [reply external/model-preparation-algorithm/mpa_tasks/apis/classification/task.py:285]: Thank you for letting me know. I expect to resolve this after separating the templates.

**Review by supersoob — APPROVED** [review]
LGTM

**Review by harimkang — APPROVED** [review]
